api/src/main/java/org/pmiops/workbench/interactiveanalysis/InteractiveAnalysisService.java

```diff
@@ -112,26 +117,31 @@ public String localize(
         isGceRuntime ? "workspaces_playground/" + workspacePath : "workspaces_playground";
     String targetDir = isPlayground ? playgroundDir : editDir;
 
+    var storageLink =
+        new StorageLink()
+            .cloudStorageDirectory(gcsNotebooksDir)
+            .localBaseDirectory(editDir)
+            .localSafeModeBaseDirectory(playgroundDir);
+
+    if (isGceRuntime) {
+      storageLink.setPattern(JUPYTER_DELOC_PATTERN);
+    } else if (AppType.RSTUDIO.equals(appType)) {
+      storageLink.setPattern(RSTUDIO_DELOC_PATTERN);
+    } else if (AppType.SAS.equals(appType)) {
```

Comment:
In my PR: https://github.com/all-of-us/workbench/pull/7979/files
I use a `Map<AppTpye, String>`. How about that structure